pkg/cloud/api/check.go

```diff
@@ -171,3 +171,41 @@ func checkSchema(t reflect.Type) error {
 
 	return nil
 }
+
+// canConvertToT checks if we can convert type From to type To. Conversion is
+// possible if type From has the subset of fields with the same type and name as
+// in To type. This function is not symmetric.
+func canConvertToT(p Path, from, to reflect.Type) error {
+	if from.Kind() != to.Kind() {
+		return fmt.Errorf("%s has different type: %v != %v", p.String(), from.Kind(), to.Kind())
+	}
+	switch from.Kind() {
+	case reflect.Pointer:
+		return canConvertToT(p.Pointer(), from.Elem(), to.Elem())
+
+	case reflect.Struct:
+		for i := 0; i < from.NumField(); i++ {
+			af := from.Field(i)
+			bf, exist := to.FieldByName(af.Name)
+			if !exist {
+				return fmt.Errorf("%s: type %T does not have field %v", p.String(), to, af.Name)
+			}
+			if err := canConvertToT(p.Field(af.Name), af.Type, bf.Type); err != nil {
```

Comment:
Is there any GCP API struct referring to the pointer of itself?

Reply:
good question :) not the ones we use in rGraph right now